web-app/src/contexts/Common.js

```diff
@@ -1,19 +1,48 @@
 import PropTypes from 'prop-types';
+import { makeStyles } from '@material-ui/core/styles';
 import { createContext, useState } from "react"
 
 export const CommonContext = createContext();
+const useStyles = makeStyles((theme) => ({
+    typography: {
+      padding: theme.spacing(),
+    },
+    closeIcon:{
+        float:"right"
+    },
+}));
 function CommonContextProvider({children}) {
     const [open, setOpen] = useState(false);
-    const [metaData,setMetaData]=useState({})
+    const [metaData,setMetaData]=useState(null);
+    const [name,setName] = useState('');
+    const [code,setCode] = useState('');
+    const [anchorEl, setAnchorEl] = useState(null);
```

Comment:
The anchor and other metadata popup variables can be stored directly in the metadata.js file I think, we can discuss it on Monday.